src/main/java/uk/gov/hmcts/ccd/domain/model/casedataaccesscontrol/matcher/LocationMatcher.java

```diff
@@ -33,9 +33,11 @@ public boolean matchAttribute(RoleAssignment roleAssignment, CaseDetails caseDet
     }
 
     private Optional<String> getLocation(CaseDetails caseDetails) {
-        JsonNode caseManagementLocation = caseDetails.getData().get(CASE_MANAGEMENT__LOCATION);
-        if (caseManagementLocation != null && caseManagementLocation.get(BASE_LOCATION) != null) {
-            return Optional.ofNullable(caseManagementLocation.get(BASE_LOCATION).asText());
+        if (caseDetails != null && caseDetails.getData() != null) {
```

Comment:
The null check should be applied well before the code comes to this method, to make sure the treatment of a null object is done correctly. Furthermore, we need to generate a warning log to flag the situations where we unexpectedly have encountered a null case object.